File: charcorba-c++/Client/src/tableau_blanc_impl.cpp
#include <string>
#include <tableau_blanc_impl.h>
#include <tableaublanc.h>
#include <client.h>

using namespace std;

TableauBlanc_impl::TableauBlanc_impl ()
{}

TableauBlanc_impl::TableauBlanc_impl (string nom_tableau)
{
	m_nom_tableau = nom_tableau ;	
}

// Implementation for interface TableauBlanc

CORBA::Boolean
TableauBlanc_impl::ajouter_pixel( const ::TableauBlanc::t_pixel& pixel )
  throw(
    ::CORBA::SystemException)
{
  CORBA::Boolean retval;
  cout << "[DEBUG]\t[Tableau - '" << m_nom_tableau << "']\tRéception d'un pixel à la position ('" << pixel[0] << "," << pixel[1] << "'" << endl ; 
  return retval; 
}

::TableauBlanc::t_pixel*
TableauBlanc_impl::get_pixel( CORBA::ULong idpixel )
  throw(
    ::CORBA::SystemException)

{
  ::TableauBlanc::t_pixel* retval = new ::TableauBlanc::t_pixel ();
  return retval; 
}

void TableauBlanc_impl::inviter_client (string pseudo)
{
	cout << "[DEBUG]\t[Tableau - '" << m_nom_tableau << "']\tOn invite le client '" << pseudo << "'" << endl ;
	CORBA::Object_var service_distant = Client::get_instance()->m_MICO_ORB->connecter_service(pseudo);

	Standard_var standard_distant = Standard::_narrow(service_distant.in()) ;
	standard_distant->inviter_client_tableaublanc(m_nom_tableau.c_str());
	m_liste_participants.insert( pair<string,Standard_var>(pseudo,standard_distant));
}

File: charcorba-java/tableau_blanc/TableauBlanc_impl.java
package tableau_blanc;

import java.util.Collection;
import java.util.HashMap;
import java.util.Iterator;
import java.util.Vector;

import org.omg.CosNaming.NamingContextPackage.CannotProceed;
import org.omg.CosNaming.NamingContextPackage.InvalidName;
import org.omg.CosNaming.NamingContextPackage.NotFound;

import Client.Standard;

public class TableauBlanc_impl extends TableauBlancPOA
{

	public String m_nom_tableau;
	public HashMap<String,Standard> m_liste_participants;
	public Vector <s_pixel> m_vect_pixels;
	
	public TableauBlanc_impl(String nom_tableau) 
	{
		m_nom_tableau = nom_tableau;
		m_liste_participants = new HashMap<String,Standard> ();
		m_vect_pixels = new Vector <s_pixel> ();
	}


	public void inviter_client(String pseudo) throws NotFound, CannotProceed, InvalidName 
	{
		org.omg.CORBA.Object service_distant = orb_pkge.COrb.static_orb.connecter_service(pseudo);;
		Client.Standard standard_distant = Client.StandardHelper.narrow(service_distant) ;
		standard_distant.inviter_client_tableaublanc(m_nom_tableau);
		m_liste_participants.put(pseudo,standard_distant);
	}


	public boolean ajouter_pixel(s_pixel pixel) 
	{
		
		  boolean retval = true;
		  System.out.println("Impl : ajout pixel "+pixel.x+" "+pixel.y);
		  m_vect_pixels.add(pixel);

          Collection<Standard> participants = m_liste_participants.values();
		  Iterator <Standard> it = participants.iterator();
		  while (it.hasNext())
		  {
					it.next().signal_tableaublanc(m_nom_tableau,m_vect_pixels.size());
		  }
		  //Client.Client.singleton_client.sync_tableau_blanc(m_nom_tableau,m_vect_pixels.size());
			
		  return retval;
	}

	public s_pixel get_pixel(int idpixel) 
	{
		return m_vect_pixels.get(idpixel);
	}

}
